src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -72,6 +72,8 @@ public interface IHostEnvironment : IChannelProvider, IProgressChannelProvider
         /// The suffix and prefix are optional. A common use for suffix is to specify an extension, eg, ".txt".
         /// The use of suffix and prefix, including whether they have any affect, is up to the host environment.
         /// </summary>
+        [Obsolete("The host environment is not disposable, so it is inappropriate to use this method. " +
+            "Handle your own temporary files. If you cannot, reconsider your life choices.")]
```

Comment:
>If you cannot, reconsider your life choices [](start = 46, length = 43)

you sure you want to show this message to a user? they might see it at 5pm on a Friday, as they are trying to wrap up their week and go home!

Reply:
Well, only component authors will run into it, and AFAIK there is only one component still that actually uses this (as seen in #1287). We are also *definitely* removing this, so an obsolete method to anyone that might make the mistake of using it in the future is I think appropriate.

Reply:
Oh you mean was I a bit too flippant... yeah a bit. It was meant to be funny but maybe it wasn't. I'll tone it down.